group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -16294,6 +16299,76 @@ barTopicName, new TopicMetadata(barTopicId, barTopicName, 3)
         );
     }
 
+    @Test
+    public void testSharePartitionKeyMap() {
+        MockPartitionAssignor assignor = new MockPartitionAssignor("range");
+        assignor.prepareGroupAssignment(new GroupAssignment(Collections.emptyMap()));
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .withConfig(GroupCoordinatorConfig.CONSUMER_GROUP_ASSIGNORS_CONFIG, List.of(assignor))
+            .build();
+
+        MetadataImage image = mock(MetadataImage.class);
+        TopicsImage topicsImage = mock(TopicsImage.class);
+        TopicImage t1image = mock(TopicImage.class);
+        TopicImage t2image = mock(TopicImage.class);
+        when(topicsImage.getTopic(anyString()))
+            .thenReturn(t1image)
+            .thenReturn(t2image);
```

Comment:
It may be easier to just build the MetadataImage that you need rather than mocking it. We have `MetadataImageBuilder` which is pretty handy for it.